### PR TITLE
Added paths via paired staBal.

### PR DESCRIPTION
**Pull request by johngrantuk**
This creates paths using staBal3 BPT as the connecting token instead of the stable tokens which was previously being used. Also maintains stable<> stable path via linear pools. Can potentially be generalised.

### CODE REVIEW

File: test/linear.spec.ts [hunk 1]
```diff
@@ -325,18 +252,37 @@ describe('linear pool tests', () => {
...
-                    ['linearUSDT', 'multiid', 'linearUSDC', 'weightedUsdcWeth'],
+                    ['weightedBalStaBal3', 'staBal3Id', 'linearDAI'],
                     poolsAllDict,
                     paths[0],
                     tokenIn,
                     tokenOut
                 );
+            });
+
+            it('should return 2 valid linear paths', async () => {
```
Comment (sergioyuhjtman) [inline test/linear.spec.ts:266]: Should it be "1 valid path" here?

File: test/linear.spec.ts [hunk 2]
```diff
@@ -325,18 +252,37 @@ describe('linear pool tests', () => {
...
+                const [paths, poolsAllDict] = getPaths(
+                    tokenIn,
+                    tokenOut,
+                    SwapTypes.SwapExactIn,
+                    testPools,
+                    maxPools
+                );
+
+                assert.equal(paths.length, 1);
+                // TokenIn>[WeightedPool]>staBal3Bpt>[staBAL3]>bStable>[LINEARPOOL]>TokenOut
```
Comment (sergioyuhjtman) [inline test/linear.spec.ts:281]: This commented path doesn't correspond to the path below

File: test/linear.spec.ts (hunk 1 above)
Comment (johngrantuk) [reply test/linear.spec.ts]: Corrected - thanks.

File: test/linear.spec.ts (hunk 2 above)
Comment (johngrantuk) [reply test/linear.spec.ts]: Corrected - thanks.

**Pull request merged**